gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -1693,6 +1646,34 @@ protected GenerateDataTemplateTask configureDataTemplateGeneration(Project proje
     prepareLegacySchemasForPublishTask.dependsOn(destroyStaleFiles);
     dataTemplateJarDepends.add(prepareLegacySchemasForPublishTask);
 
+    // extension schema directory
+    File extensionSchemaDir = project.file(getExtensionSchemaPath(project, sourceSet));
+
+    if (!SharedFileUtils.getSuffixedFiles(project, extensionSchemaDir, PDL_FILE_SUFFIX).isEmpty()) {
+      // Validate extension schemas if extension schemas are provided.
+      ValidateExtensionSchemaTask validateExtensionSchemaTask = project.getTasks()
+          .create(sourceSet.getTaskName("validate", "ExtensionSchema"), ValidateExtensionSchemaTask.class, task -> {
```

Comment:
It would make more sense if it was called `validateExtensionSchemas` IMO

Reply:
fixed